adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Telemetry.cs

```diff
@@ -0,0 +1,56 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.Identity.Core.Telemetry;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal class Telemetry : ITelemetry
+    {
+        private static readonly ITelemetry Instance = new Telemetry();
+
+        public static Telemetry GetInstance()
+        {
+            return Instance as Telemetry;
+        }
+
+        void ITelemetry.StartEvent(string requestId, EventBase eventToStart)
```

Comment:
>ITelemetry [](start = 13, length = 10)

why explicitly?

Reply:
the class is internal, it will not be public. This should not have anything to do with the interface being implicitly implemented. I don't get why it has to be public to be explicitly implemented

---
In reply to: [209065629](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1173#discussion_r209065629) [](ancestors = 209065629)

Reply:
@henrik-me Ah, I understand, I confused this with the other Telemetry public class in MSAL, you are right, it does not need to be implemented explicitly. I will remove